server/src/main/java/org/elasticsearch/index/fielddata/SortableLongBitsSortedNumericDocValues.java

```diff
@@ -45,4 +47,11 @@ public int docValueCount() {
     public SortedNumericDoubleValues getDoubleValues() {
         return values;
     }
+
+    @Override
+    public void setScorer(Scorable scorer) {
+        if (values instanceof ScorerAware aware) {
+            aware.setScorer(scorer);
+        }
```

Comment:
This is the actual bug fix.

Reply:
ooo, the new instance of syntax.  Very nice.